src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    [EventSource(
+        Guid = "20b30044-b729-457e-8dda-8b41b5dd02e6", 
+        Name = "System.Buffers.BufferPoolEventSource",
+        LocalizationResources = "FxResources.System.Buffers.SR")]
+    internal sealed class ArrayPoolEventSource : EventSource
+    {
+        internal readonly static ArrayPoolEventSource Log = new ArrayPoolEventSource();
+
+        internal enum BufferAllocationReason : int
+        {
+            Pooled,
+            OverMaximumSize,
+            PoolExhausted
+        }
+
+        [Event(1, Level = EventLevel.Informational)]
+        internal void BufferRented(int bufferId, int bufferSize, int poolId, int bucketId)
+        {
+             unsafe
+             {
+                 EventData* payload = stackalloc EventData[4];
+                 payload[0].Size = sizeof(int);
+                 payload[0].DataPointer = ((IntPtr) (&bufferId));
+                 payload[1].Size = sizeof(int);
+                 payload[1].DataPointer = ((IntPtr) (&bufferSize));
+                 payload[2].Size = sizeof(int);
+                 payload[2].DataPointer = ((IntPtr) (&poolId));
+                 payload[3].Size = sizeof(int);
+                 payload[3].DataPointer = ((IntPtr) (&bucketId));
+                 WriteEventCore(1, 4, payload);
+             }
```

Comment:
You could move this out into your own WriteEvent and then use it in both BufferRented and BufferAllocated, e.g.
```C#
private unsafe void WriteEvent(int eventId, int arg0, int arg1, int arg2, int arg3)
{
    if (IsEnabled())
    {
        unsafe
        {
            EventData* payload = stackalloc EventData[4];
            payload[0].Size = sizeof(int);
            payload[0].DataPointer = (IntPtr) (&arg0);
            EventData* payload = stackalloc EventData[4];
            payload[1].Size = sizeof(int);
            payload[1].DataPointer = (IntPtr) (&arg1);
            EventData* payload = stackalloc EventData[4];
            payload[2].Size = sizeof(int);
            payload[2].DataPointer = (IntPtr) (&arg2);
            EventData* payload = stackalloc EventData[4];
            payload[3].Size = sizeof(int);
            payload[3].DataPointer = (IntPtr) (&arg3);
            WriteEventCore(1, 4, payload);
        }
    }
}
```